### ISSUE TITLE
Shaper mod not calculated

### ISSUE THREAD

**Issue by mickeymicky2** [Openarl/PathOfBuilding#803]
When you have a Shaper item as offhand with "gain x% of physical damage as extra damage of a random element", it's DPS is not calculated. 

But it calculates the other mods such as the "gain x% as extra cold/fire/chaos/lightning damage".

**Comment by livejamie** [comment]
Intelligence % on Body Armors don't seem to be taken into account as well

**Comment by Openarl** [comment]
Unfortunately, correctly simulating the addition of a random element is quite difficult to do, and beyond the program's current capabilities; it may be a while before I'm able to add support for it.

As for +% increased Intelligence, as far as I can tell it is working normally. I'd be very surprised if it wasn't, as it is such a basic stat.

**Comment by SeventhHelix** [comment]
I ran into this same 'random element' limitation for a gear upgrade and took a look to see if I could make a change myself.

@Openarl would it be possible to do a simple calculation by calculating `gain x% of physical damage as extra [fire, cold, lightning] damage` and averaging them? I'm not sure how accurate this would be, but it's roughly what I do when checking items manually.

**Comment by snuze** [comment]
I think the implementation suggested by SeventHelix makes the most sense. Please consider it!

**Comment by RandomHero13** [comment]
yes, that would be a (rough) estimate, which i think is better than nothing.
